src/test/ui/associated-types/defaults-unsound-62211-1.rs

```diff
@@ -0,0 +1,62 @@
+//! Regression test for https://github.com/rust-lang/rust/issues/62211
```

Comment:
I'll note that the root cause of #62211 is the same as #33017 - wfcheck can rely on the projection predicate it is itself checking, and that still isn't fixed.

Reply:
^--- should be added as a comment in the regression test. :)

Reply:
I added the comment.

> I'll note that the root cause of #62211 is the same as #33017

Do you think that this is still an issue without using specialization (ie. just using associated type defaults as implemented by this PR)?